### ISSUE TITLE
Dropping nullability of struct field of unconstrained type

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#32338]
No warning should be reported for `t.F.ToString()`:
```C#
#pragma warning disable 649

struct S<T>
{
    internal T F;
}

class Program
{
    static void M<T>(S<T> s)
    {
        if (s.F == null) return;
        var t = s;
        t.F.ToString(); // warning: maybe null
    }
}
```
